### PR TITLE
Servlets: books catalog

### PR DESCRIPTION
**Pull request by slavidichkov**

### CODE REVIEW

File: bookscatalog/src/main/java/com/clouway/adapter/inmemory/InmemoryBooksRepository.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        add(new Book("database", "Helikon", 1213243535L));
+        add(new Book("PHP", "Arhimed", 1213243535L));
+        add(new Book("AJAX", "Anubis", 1213243535L));
+        add(new Book("CSS", "Helikon", 1213243535L));
+        add(new Book("Go", "Miranda", 1213243535L));
+        add(new Book("Goos", "Helikon", 1213243535L));
+        add(new Book("Java clean code", "Bullvest", 1213243535L));
+        add(new Book("Design paterns", "Helikon", 1213243535L));
+    }};
+    public List<Book> get() {
```
Comment (tsony-tsonev) [inline bookscatalog/src/main/java/com/clouway/adapter/inmemory/InmemoryBooksRepository.java]: bad name

File: bookscatalog/src/main/java/com/clouway/core/PageBean.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package com.clouway.core;
+
+import java.util.List;
+
+/**
+ * @author Slavi Dichkov (slavidichkof@gmail.com)
+ */
+public interface PageBean<T> {
+    void setList(List<T> list);
```
Comment (tsony-tsonev) [inline bookscatalog/src/main/java/com/clouway/core/PageBean.java]: bad name

File: bookscatalog/src/main/java/com/clouway/core/PageBeanImpl.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import java.util.List;
+
+/**
+ * @author Slavi Dichkov (slavidichkof@gmail.com)
+ */
+public class PageBeanImpl<T> implements PageBean{
+    private int firstIndex;
+    private int lastIndex;
+    private int pageSize = 10;
+    private List<T> rows = null;
```
Comment (tsony-tsonev) [inline bookscatalog/src/main/java/com/clouway/core/PageBeanImpl.java]: probably items will be better (cause you'll have items in page, and rows is used in tables)

File: bookscatalog/src/main/java/com/clouway/adapter/inmemory/InmemoryBooksRepository.java (hunk 1 above)
Comment (slavidichkov) [inline bookscatalog/src/main/java/com/clouway/adapter/inmemory/InmemoryBooksRepository.java]: fixed

File: bookscatalog/src/main/java/com/clouway/core/PageBeanImpl.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+    public void setList(List list) {
+        if (rows==null){
```
Comment (tsony-tsonev) [inline bookscatalog/src/main/java/com/clouway/core/PageBeanImpl.java]: this way you can not change the list once it is set cause on the second try it won't be null

File: bookscatalog/src/main/java/com/clouway/core/PageBeanImpl.java (hunk 3 above)
Comment (slavidichkov) [inline bookscatalog/src/main/java/com/clouway/core/PageBeanImpl.java]: fixed

File: bookscatalog/src/main/java/com/clouway/core/PageBean.java (hunk 2 above)
Comment (slavidichkov) [inline bookscatalog/src/main/java/com/clouway/core/PageBean.java]: fixed

**Comment by tsony-tsonev** [comment]
LGTM

**Pull request merged**